### PR TITLE
feat: update GetAuthorizationUri, ConnectorSession and social connector member methods

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. Pass parameters `connectorId`, `connectorFactoryId` and `jti` to `GetAuthorizationUri`.
2. Make `ConnectorSession` compatible with arbitrary keys.
3. Add the optional `validateSamlAssertion` method and its type definition to social connectors for the SAML connector's use case.

## Testing

N/A

### CODE REVIEW

**Review by charIeszhao — APPROVED** [review]

File: packages/core/src/routes/interaction/utils/social-verification.ts [hunk 1]
```diff
@@ -33,7 +33,18 @@ export const createSocialAuthorizationUrl = async (
...
+    {
+      state,
+      redirectUri,
+      /**
+       * For upcoming POST /interaction/verification/assertion API, we need to block requests for non-SAML connector (relies on connectorFactoryId)
+       * and use `connectorId` to find correct connector config.
+       *
+       * TODO @darcy : add check on `connectorId` and `connectorFactoryId` existence and save logic in SAML connector `getAuthorizationUri` method.
+       */
+      customConfigs: { connectorId, connectorFactoryId: connector.metadata.id },
```
Comment (wangsijie) [inline packages/core/src/routes/interaction/utils/social-verification.ts:45]: can we remove `customConfigs` and move `connectorId` and `connectorFactoryId` to the parent arguments?

File: packages/toolkit/connector-kit/src/types.ts [hunk 2]
```diff
@@ -162,13 +172,24 @@ export type SendMessageFunction = (
 export type SocialConnector = BaseConnector<ConnectorType.Social> & {
   getAuthorizationUri: GetAuthorizationUri;
   getUserInfo: GetUserInfo;
+  validateAssertion?: ValidateAssertion;
 };
 
+// This type definition is for SAML connector
+export type ValidateAssertion = (
```
Comment (simeng-li) [inline packages/toolkit/connector-kit/src/types.ts:179]: rename as SAMLValidateAssertion ?

**Review by wangsijie — APPROVED** [review]

File: .changeset-staged/good-feet-own.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@logto/connector-kit": minor
```
Comment (gao-sun) [inline .changeset-staged/good-feet-own.md:2]: should be patch as it has no new features?

File: packages/toolkit/connector-kit/src/types.ts (hunk 2 above)
Comment (darcyYe) [reply packages/toolkit/connector-kit/src/types.ts]: Renamed to ValidateSamlAssertion to follow v. + n. naming convention

File: .changeset-staged/good-feet-own.md (hunk 3 above)
Comment (darcyYe) [reply .changeset-staged/good-feet-own.md]: updated

**Review by simeng-li — APPROVED** [review]

**Pull request merged**